pkg/kube/inject/inject.go

```diff
@@ -1111,36 +1115,42 @@ func rewriteCniPodSpec(annotations map[string]string, spec *SidecarInjectionSpec
 
 // overwriteClusterInfo updates cluster name and network from url path
 // This is needed when webconfig config runs on a different cluster than webhook
+// follow format: /inject/k1/v1/k2/v2, any kv order works
+// eg. "/inject/cluster/cluster1", "/inject/net/network1/cluster/cluster1"
 func overwriteClusterInfo(containers []corev1.Container, path string) {
 	res := strings.Split(path, "/")
-	if len(res) >= 5 {
-		// if len is less than 5, not enough length for /cluster/X/net/Y
-		clusterName, clusterNetwork := "", ""
-		clusterName = res[len(res)-3]
-		clusterNetwork = res[len(res)-1]
+	newEnvs := make(map[string]string)
 
-		log.Debugf("Updating cluster info based on clusterName: %s clusterNetwork: %s\n", clusterName, clusterNetwork)
+	for i := 2; i+1 < len(res); i += 2 { // skip '/inject'
```

Comment:
nit: can we break this out and add unit test? Also probably need an error/warning for an odd number of entries to avoid things like `/inject/foo`

Reply:
Error would be a log probably

Reply:
1. unit tests added
2. add warn log if odd kv found
@howardjohn